composer/core/evaluator.py

```diff
@@ -35,13 +36,14 @@ def evaluate_periodically(eval_interval: Union[str, Time, int], eval_at_fit_end:
     if isinstance(eval_interval, str):
         eval_interval = Time.from_timestring(eval_interval)
 
-    if eval_interval.unit not in (TimeUnit.EPOCH, TimeUnit.BATCH):
-        raise ValueError('The `eval_interval` must have units of EPOCH or BATCH, or be a function.')
+    if eval_interval.unit not in (TimeUnit.EPOCH, TimeUnit.BATCH, TimeUnit.DURATION):
+        raise ValueError('The `eval_interval` must have units of EPOCH, BATCH, DURATION or be a function.')
 
     last_batch_seen = -1
 
     def should_eval(state: State, event: Event):
-        if int(eval_interval) <= 0:
+        # `TimeUnit.Duration` value is a floating value and ranges from `[0.0, 1.0)`
```

Comment:
```suggestion
        # `TimeUnit.Duration` value is a float from `[0.0, 1.0)`
```

Reply:
Done in https://github.com/mosaicml/composer/pull/1523/commits/7fb861f1fd1c6b4461132c9092ec826f48a00071.